compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -788,3 +788,64 @@ macro_rules! define_queries {
         }
     }
 }
+
+pub fn print_query_stack<'tcx>(
+    qcx: QueryCtxt<'tcx>,
+    mut current_query: Option<QueryJobId>,
+    handler: &Handler,
+    num_frames: Option<usize>,
+) -> usize {
+    // Be careful relying on global state here: this code is called from
+    // a panic hook, which means that the global `Handler` may be in a weird
+    // state if it was responsible for triggering the panic.
+    let mut i = 0;
+
+    #[cfg(not(parallel_compiler))]
+    let query_map = qcx.try_collect_active_jobs();
+
+    #[cfg(parallel_compiler)]
+    let query_map = rustc_middle::ty::tls::with_context(|context| {
```

Comment:
Could you add a comment explaining why we spawn a thread here?

Reply:
Done :)